luarules/gadgets/unit_dgun_behaviour.lua

```diff
@@ -11,11 +11,22 @@ end
 if not gadgetHandler:IsSyncedCode() then
 	return
 end
+
+local spGetProjectileDefID = Spring.GetProjectileDefID
+local spGetUnitShieldState = Spring.GetUnitShieldState
+local spGetUnitDefID = Spring.GetUnitDefID
+local spSetUnitShieldState = Spring.SetUnitShieldState
+local diag = math.diag
+
+
+local dgunWeaponsTTL = {}
 local dgunWeapons = {}
 for weaponDefID, weaponDef in ipairs(WeaponDefs) do
 	if weaponDef.type == 'DGun' then
+		Spring.Echo(weaponDef)
 		Script.SetWatchProjectile(weaponDefID, true)
 		dgunWeapons[weaponDefID] = weaponDef
+		dgunWeaponsTTL[weaponDefID] = math.ceil((weaponDef.range/weaponDef.projectilespeed)*30)
```

Comment:
```suggestion
		dgunWeaponsTTL[weaponDefID] = math.ceil((weaponDef.range/weaponDef.projectilespeed)*Game.gameSpeed)
```

Reply:
I've since moved onto a localized counter method that increments upon gameframe. Is this still necessary? What does it do and why do I see it in so many gadgets?